### ISSUE TITLE
Some Python exception classes not highlighted (e.g. BlockingIOError)

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#795]
(Original issue 1088 created by vrutsky on 2015-02-25T18:43:24.197912+00:00)

See https://docs.python.org/3.4/library/exceptions.html#exception-hierarchy as an example of not highlighted builtin exceptions.

Following exceptions from that page are not highlighted:

```python
BlockingIOError
ChildProcessError
ConnectionError
BrokenPipeError
ConnectionAbortedError
ConnectionRefusedError
ConnectionResetError
FileExistsError
FileNotFoundError
InterruptedError
IsADirectoryError
NotADirectoryError
PermissionError
ProcessLookupError
TimeoutError
ResourceWarning
```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2015-03-08T06:47:08.193953+00:00)

These exceptions are highlighted, but only in the Python3 lexer, since they are exclusive to 3.3+.

**Comment by Anteru** [comment]
(Original comment by vrutsky on 2015-03-08T10:36:44.329237+00:00)

Argh, looks like I searched Pygments sources for [ResourceWarning](https://docs.python.org/3.4/library/exceptions.html#ResourceWarning) (new in Pythnon 3.2), didn't find it, and decided that all other Python 3 exceptions are not handled too.

I still don't see that `ResourceWarning` is handled in sources, and running `pygmentize -l python3 ...` doesn't highlight it.
@birkenfeld can you check that `ResourceWarning` is (not) handled?

Also strange that in official Python docs so much exceptions are not highlighted, since `highlight_language = 'python3'` in their Sphinx configuration: https://hg.python.org/cpython/file/41ce95a5b2d8/Doc/conf.py#l34
Probably they some ancient version of Pygments.

**Comment by Anteru** [comment]
(Original comment by vrutsky on 2015-03-08T10:38:24.152311+00:00)

Looks like only `ResourceWarning` exception is not handled (even with `python3` lexer).

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2015-03-08T14:13:00.008436+00:00)

Closes #1088: add ResourceWarning.

→ <<cset 5ba5d9a17bb5>>
